privacyidea/lib/utils/compare.py

```diff
@@ -26,21 +26,46 @@
  2) implement a comparison function and add it to COMPARATOR_FUNCTIONS
  3) add a description of the comparator to COMPARATOR_DESCRIPTIONS
 """
+import csv
 import logging
+import re
+from six import wraps
 
 from privacyidea.lib.framework import _
 
 log = logging.getLogger(__name__)
 
 
 class CompareError(Exception):
+    """
+    Signals that an error occurred when carrying out a comparison.
+    The error message is not presented to the user, but written to the logfile.
+    """
     def __init__(self, message):
         self.message = message
 
     def __repr__(self):
         return u"CompareError({!r})".format(self.message)
 
 
+def parse_comma_separated_string(input_string):
+    """
+    Parse a string that contains a list of comma-separated values and return the list of values.
+    Each value may be quoted with a doublequote, and doublequotes may be escaped with a backslash.
+    Whitespace immediately following a delimiter is skipped.
+    Raise a CompareError if the input is malformed.
+    :param input_string: an input string
+    :return: a list of strings
+    """
+    # We use Python's csv module because it supports quoted values
+    try:
+        reader = csv.reader([input_string], strict=True, skipinitialspace=True, doublequote=False, escapechar="\\")
+        rows = list(reader)
+    except Exception as exx:
```

Comment:
can we have a more explicit Exception here? What does the `csv.reader` throw in case of an error?

Reply:
Fair point :-) Fixed this in 9c93c15aa1f1c2118cd5bfa60828364001537b7b